lib/dataset/dataset.cpp

```diff
@@ -278,8 +278,8 @@ typename Masks::holder_type union_or(const Masks &currentMasks,
     if (!currentMasks.contains(key))
       out.insert_or_assign(key, copy(item));
     else if (item.dtype() != core::dtype<bool>) {
```

Comment:
As you just mentioned, we should also check the unit here, as boolean operations are not supported for variables with a unit.